ui/shared/AssetAndAmountInput.qml

```diff
@@ -0,0 +1,146 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+import QtGraphicalEffects 1.13
+import "../imports"
+
+Item {
+  property var assets
+  property string errorMessage: ""
+  property string defaultCurrency: "USD"
+  property string fiatBalance: "0.00"
+  property alias text: inputAmount.text
+  signal inputChanged(input: string, assetSymbol: string, currency: string)
+  signal fiatValueChanged(input: string, assetSymbol: string, currency: string)
+
+  id: root
+
+  height: inputAmount.height + txtFiatBalance.height + txtFiatBalance.anchors.topMargin
+  anchors.right: parent.right
+  anchors.left: parent.left
+
+  onAssetsChanged: {
+    txtBalance.text = selectAsset.selectedAsset.value
+  }
+
+  Item {
+      anchors.right: parent.right
+      anchors.left: parent.left
+      anchors.top: parent.top
+      height: txtBalanceDesc.height
+
+      StyledText {
+          id: txtBalanceDesc
+          text: qsTr("Balance: ")
+          anchors.right: txtBalance.left
+          font.weight: Font.Medium
+          font.pixelSize: 13
+          color: parseFloat(inputAmount.text) > parseFloat(txtBalance.text) ? Style.current.red : Style.current.grey
+      }
+
+      StyledText {
+          id: txtBalance
+          text: selectAsset.selectedAsset.value
+          anchors.right: parent.right
+          font.weight: Font.Medium
+          font.pixelSize: 13
+          color: parseFloat(inputAmount.text) > parseFloat(txtBalance.text) ? Style.current.red : Style.current.grey
+
+          MouseArea {
+              cursorShape: Qt.PointingHandCursor
+              anchors.fill: parent
+              hoverEnabled: true
+              onExited: {
+                  txtBalance.color = parseFloat(inputAmount.text) > parseFloat(txtBalance.text) ? Style.current.red : Style.current.grey
+              }
+              onEntered: {
+                  txtBalance.color = Style.current.textColor
+              }
```

Comment:
The `onEntered` and `onExited` handlers don't seem to work consistently... @jrainville maybe you have an idea what's up here?

The is is the following: the text color of `txtBalance` needs to turn red when the amount value is insufficient balance and stay grey when it's fine. However, during hover, it should turn black (as an indication that it can be clicked).

I noticed that, once hovered or clicked, it will never change it's color again, even thought the amount value might be invalid (the text "Balance: " next to this one behaves properly on the other hand)

Reply:
One thing you could do is instead of changing the color in those functions, try to add a `hovered` property to the `txtBalance` and let `txtBalance` handle its color alone depending on `hovered` and on the balance

Reply:
Jonathan’s suggestion is also required to avoid “remembering” of the previous colour after changing themes as well.